src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -417,8 +417,8 @@ private function getHiddenResponse(OutputInterface $output, $inputStream, bool $
                 $exe = $tmpExe;
             }
 
-            $sExec = shell_exec('"'.$exe.'"');
-            $value = $trimmable ? rtrim($sExec) : $sExec;
+            $sExec = exec('"'.$exe.'"');
+            $value = $trimmable ? rtrim($sExec) : $sExec . PHP_EOL; // add PHP_EOL to keep the same result as shell_exec
```

Comment:
Do we need this or can this be ignored @nicolas-grekas ?

Reply:
I don't think so as `rtrim()` would have removed it before this patch

Reply:
I played a bit with exec vs shell_exec and what the php doc says is correct: lines are trimmed when using exec.
That means using it would break answers with terminal spaces.
The only way is using shell_exec indeed.
What about doing the reverse and always using shell_exec everywhere, including in the function_exists check?

Reply:
👍 exec would break multiline answers as well since it only returns the last line

Reply:
@nicolas-grekas in `QuestionHelper::isInteractiveInput` for example we use `exec` to also get the result code.

Reply:
I think we can use $output instead: it's going to be empty if stty fails:
\'' === trim(shell_exec(...))\